src/app/functions/server/getExperiments.ts

```diff
@@ -28,9 +28,22 @@ export async function getExperiments(params: {
   countryCode: string;
   previewMode: boolean;
 }): Promise<ExperimentData> {
+  // TODO MNTOR-3380 - until Cirrus implements preview mode, set all Nimbus features to `true` for QA purposes.
+  if (params.previewMode === true) {
+    // Clone the `localExperimentData` object so we don't modify the exported data structure.
+    const overriddenExperimentData = localExperimentData;
+    Object.assign(overriddenExperimentData, localExperimentData);
+    Object.keys(overriddenExperimentData).forEach((key) => {
+      const feature = overriddenExperimentData[key as keyof ExperimentData];
+      feature["enabled"] = true;
+    });
+    return overriddenExperimentData;
```

Comment:
I'm not sure how to reproduce the problem you're describing in the PR, but I don't think this does what you want, as the properties still reference the same objects, i.e. even though `overriddenExperimentData !== localExperimentData`, it's still the case that `overriddenExperimentData["last-scan-date"] === localExperimentData["last-scan-date"]`. I think you want something like this:

```suggestion
    const overriddenExperimentData = Object.fromEntries(
      Object.entries(localExperimentData).map(
        ([experimentId, experimentConfig]) => {
          return [experimentId, { ...experimentConfig, enabled: true }];
        },
      ),
    ) as ExperimentData;
    return overriddenExperimentData;
```

Reply:
Thanks! I think https://developer.mozilla.org/en-US/docs/Web/API/structuredClone would do the right thing as well.

The problem I noticed is that modifications to the `localExperimentData` export from the generated Nimbus module seem to be cached for the lifetime of the application, but if `Object.assign` is keeping any references to the old object then that's true it's not going to solve much.